File: flutter/lib/src/cheyalpaadugal/niril_pattu/kaatchi/koorugal/pattu_mugappu_attai.dart
import 'package:flutter/material.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:intl/intl.dart';

import '../../../../adippadai/tharavuru/uruvugal.dart';
import '../../../../adippadai/mozhiyaakkam/mozhi_vazhanguthi.dart';

/// Pixel-perfect port of React's ElvanCard + renderRecentItem for Silk mode.
/// Shows: index circle + customer name + invoice # + date + amount.
class PattuMugappuAttai extends ConsumerStatefulWidget {
  const PattuMugappuAttai({
    super.key,
    required this.index,
    required this.pattiyal,
    required this.onTap,
  });

  final int index;
  final PattiyalTharavuru pattiyal;
  final VoidCallback onTap;

  @override
  ConsumerState<PattuMugappuAttai> createState() => _PattuMugappuAttaiState();
}

class _PattuMugappuAttaiState extends ConsumerState<PattuMugappuAttai> {
  bool _isPressed = false;
  static final _dateFormat = DateFormat('dd/MM/yyyy');
  static final _currencyFormat =
      NumberFormat.currency(locale: 'en_IN', symbol: '₹');

  @override
  Widget build(BuildContext context) {
    final isDark = Theme.of(context).brightness == Brightness.dark;
    final p = widget.pattiyal;
    final amountStr = _currencyFormat.format(p.mothaThogai);

    final currentLocale = ref.watch(localeProvider);
    final effectiveLang =
        currentLocale?.languageCode ?? Localizations.localeOf(context).languageCode;

    final primaryLang = effectiveLang == 'ta' ? 'Tamil' : 'English';
    final secondaryLang = effectiveLang == 'ta' ? 'English' : 'Tamil';

    return GestureDetector(
      onTapDown: (_) => setState(() => _isPressed = true),
      onTapUp: (_) => setState(() => _isPressed = false),
      onTapCancel: () => setState(() => _isPressed = false),
      child: AnimatedScale(
        scale: _isPressed ? 0.985 : 1.0,
        duration: _isPressed
            ? const Duration(milliseconds: 100)
            : const Duration(milliseconds: 200),
        curve: Curves.easeOut,
        child: ClipRRect(
          borderRadius: BorderRadius.circular(24),
          child: Material(
            color: isDark ? const Color(0xFF111111) : Colors.white,
            child: InkWell(
              borderRadius: BorderRadius.circular(24),
              onTap: widget.onTap,
              child: Padding(
                padding: const EdgeInsets.all(16),
                child: Row(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    // Index circle
                    Container(
                      width: 28,
                      height: 28,
                      margin: const EdgeInsets.only(top: 1),
                      decoration: BoxDecoration(
                        color: isDark
                            ? Colors.white.withValues(alpha: 0.12)
                            : Colors.black.withValues(alpha: 0.08),
                        shape: BoxShape.circle,
                      ),
                      child: Center(
                        child: Text(
                          (widget.index + 1).toString().padLeft(2, '0'),
                          style: TextStyle(
                            fontSize: 11.2,
                            fontWeight: FontWeight.w800,
                            color: isDark ? Colors.white : Colors.black,
                            height: 1,
                          ),
                        ),
                      ),
                    ),
                    const SizedBox(width: 12),
                    // Content
                    Expanded(
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        mainAxisSize: MainAxisSize.min,
                        children: [
                          // Row 1: Customer name + chevron
                          Row(
                            children: [
                              Expanded(
                                child: Text(
                                  p.vaangunarPeyar[primaryLang]?.isNotEmpty == true
                                      ? p.vaangunarPeyar[primaryLang]!
                                      : p.vaangunarPeyar[secondaryLang] ?? '-',
                                  style: const TextStyle(
                                    fontSize: 15.2,
                                    fontWeight: FontWeight.w700,
                                  ),
                                  maxLines: 1,
                                  overflow: TextOverflow.ellipsis,
                                ),
                              ),
                              Icon(
                                CupertinoIcons.chevron_right,
                                size: 18,
                                color: isDark
                                    ? const Color(0xFF555555)
                                    : const Color(0xFFAAAAAA),
                              ),
                            ],
                          ),
                          // Row 2: Invoice # + Date
                          const SizedBox(height: 4),
                          Text.rich(
                            TextSpan(
                              children: [
                                TextSpan(text: p.patrucheettuEn),
                                TextSpan(
                                  text: '  •  ',
                                  style: TextStyle(
                                    color: (isDark ? Colors.white : Colors.black)
                                        .withValues(alpha: 0.4),
                                  ),
                                ),
                                TextSpan(text: _dateFormat.format(p.pattiyalNaal)),
                              ],
                            ),
                            style: TextStyle(
                              fontSize: 13.6,
                              color: isDark ? Colors.white54 : Colors.black54,
                            ),
                            maxLines: 1,
                            overflow: TextOverflow.ellipsis,
                          ),
                          // Row 3: Amount (right-aligned)
                          const SizedBox(height: 2),
                          Align(
                            alignment: Alignment.centerRight,
                            child: Text(
                              amountStr,
                              style: TextStyle(
                                fontSize: amountStr.length > 11 ? 12.8 : 15.2,
                                fontWeight: FontWeight.w800,
                                color: Theme.of(context).colorScheme.primary,
                              ),
                            ),
                          ),
                        ],
                      ),
                    ),
                  ],
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }
}

File: migrate/composeApp/src/commonMain/kotlin/com/elvan/noolachu/ui/screens/home/koorugal/PattuMugappuAttai.kt
package com.elvan.noolachu.ui.screens.home.koorugal

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.elvan.noolachu.core.utils.CurrencyUtils
import com.elvan.noolachu.core.utils.DateUtils
import com.elvan.noolachu.data.model.PattiyalTharavuru
import com.elvan.noolachu.localization.LocalAppLanguage
import com.elvan.noolachu.theme.LocalAppFontFamily
import com.elvan.noolachu.theme.ShellColors
import com.elvan.noolachu.theme.preventBrokenLigatures
import com.elvan.noolachu.ui.components.ElvanPothuAttai
import com.elvan.noolachu.ui.navigation.MaterialSymbols

/**
 * Pixel-perfect port of Flutter's PattuMugappuAttai.
 */
@Composable
fun PattuMugappuAttai(
    index: Int,
    pattiyal: PattiyalTharavuru,
    colors: ShellColors,
    modifier: Modifier = Modifier,
    onClick: () -> Unit = {}
) {
    val ff = LocalAppFontFamily.current
    val isDark = colors.isDark
    val currentLang = LocalAppLanguage.current

    val name = pattiyal.vaangunarPeyar[currentLang]
        ?: pattiyal.vaangunarPeyar["ta"]
        ?: pattiyal.vaangunarPeyar["en"]
        ?: pattiyal.vaangunarPeyar.values.firstOrNull()
        ?: "-"

    val amountStr = CurrencyUtils.formatInr(pattiyal.mothaThogai)
    val dateStr = DateUtils.formatEpochMillis(pattiyal.pattiyalNaal)

    ElvanPothuAttai(
        onClick = onClick,
        modifier = modifier,
        padding = PaddingValues(16.dp),
        borderRadius = 24.dp
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.Top
        ) {
            // Index circle (28x28)
            Box(
                modifier = Modifier
                    .padding(top = 1.dp)
                    .size(28.dp)
                    .clip(CircleShape)
                    .background(
                        if (isDark) Color.White.copy(alpha = 0.12f)
                        else Color.Black.copy(alpha = 0.08f)
                    ),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = (index + 1).toString().padStart(2, '0'),
                    style = TextStyle(
                        fontFamily = ff,
                        fontSize = 11.2.sp,
                        fontWeight = FontWeight.ExtraBold,
                        color = if (isDark) Color.White else Color.Black,
                        lineHeight = 11.2.sp
                    )
                )
            }

            Spacer(modifier = Modifier.width(12.dp))

            // Content
            Column(
                modifier = Modifier.weight(1f)
            ) {
                // Row 1: Customer Name + Chevron
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        text = name.preventBrokenLigatures(),
                        style = TextStyle(
                            fontFamily = ff,
                            fontSize = 15.5.sp,
                            fontWeight = FontWeight.Bold,
                            color = colors.textPrimary
                        ),
                        modifier = Modifier.weight(1f),
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis
                    )

                    Icon(
                        imageVector = MaterialSymbols.Rounded.ChevronRight,
                        contentDescription = null,
                        tint = if (isDark) Color(0xFF555555) else Color(0xFFAAAAAA),
                        modifier = Modifier.size(18.dp)
                    )
                }

                Spacer(modifier = Modifier.height(4.dp))

                // Row 2: Invoice #  •  Date
                Text(
                    text = "${pattiyal.patrucheettuEn}  •  $dateStr",
                    style = TextStyle(
                        fontFamily = ff,
                        fontSize = 13.5.sp,
                        color = if (isDark) Color.White.copy(alpha = 0.54f) else Color.Black.copy(alpha = 0.54f)
                    ),
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )

                Spacer(modifier = Modifier.height(4.dp))

                // Row 3: Right-aligned Amount
                Text(
                    text = amountStr,
                    style = TextStyle(
                        fontFamily = ff,
                        fontSize = if (amountStr.length > 11) 13.sp else 15.5.sp,
                        fontWeight = FontWeight.ExtraBold,
                        color = colors.accent
                    ),
                    modifier = Modifier.fillMaxWidth(),
                    textAlign = TextAlign.End
                )
            }
        }
    }
}
